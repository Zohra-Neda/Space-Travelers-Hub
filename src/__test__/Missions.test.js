import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import Missions from '../pages/Missions';
// Create a mock store using redux-mock-store
const mockStore = configureMockStore();
const store = mockStore({
  missions: {
    value: [
      {
        id: 1, mission: 'Mission 1', description: 'Mission 1 Description', status: 'Active',
      },
      {
        id: 2, mission: 'Mission 2', description: 'Mission 2 Description', status: 'Inactive',
      },
    ],
  },
});

describe('Missions', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
