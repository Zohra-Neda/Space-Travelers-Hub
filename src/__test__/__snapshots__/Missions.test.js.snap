// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component renders correctly 1`] = `
<table
  className="mission-container border border-1 w-[80%] mx-auto my-10"
>
  <tr
    className="header-table"
  >
    <th>
      Mission
    </th>
    <th>
      Description
    </th>
    <th>
      Status
    </th>
    <th>
      Join/Leave
    </th>
  </tr>
  <tr>
    <th
      className=" w-[10%]"
    >
      Mission 1
    </th>
    <th
      className="description-row"
    >
      This is mission 1 description
    </th>
    <th
      className=" w-[17%]"
    >
      <span
        className=" bg-gray-400 rounded-md px-2 py-1"
      >
        Not a Member
      </span>
    </th>
    <th
      className=" w-[15%]"
    >
      <button
        aria-label="Reserve Rocket"
        className=" px-1 py-1 border border-slate-700 rounded-md text-gray-500"
        onClick={[Function]}
        type="button"
      >
        Join Meeting
      </button>
    </th>
  </tr>
  <tr>
    <th
      className=" w-[10%]"
    >
      Mission 2
    </th>
    <th
      className="description-row"
    >
      This is mission 2 description
    </th>
    <th
      className=" w-[17%]"
    >
      <span
        className="bg-green-500 text-white rounded-md px-1"
      >
        Active Member
      </span>
    </th>
    <th
      className=" w-[15%]"
    >
      <button
        aria-label="Reserve Rocket"
        className="border px-2 py-1 rounded-md border-red-500 text-red-500"
        onClick={[Function]}
        type="button"
      >
        Leave Meeting 
      </button>
    </th>
  </tr>
</table>
`;
